crates/compiler/builtins/bitcode/src/str.zig

```diff
@@ -1878,8 +1878,47 @@ fn errorToProblem(bytes: [*]u8, length: usize) struct { index: usize, problem: U
 }
 
 pub fn isValidUnicode(ptr: [*]u8, len: usize) callconv(.C) bool {
-    const bytes: []u8 = ptr[0..len];
-    return @call(.{ .modifier = always_inline }, unicode.utf8ValidateSlice, .{bytes});
+    const buf: []u8 = ptr[0..len];
+
+    const size = @sizeOf(u64);
+    // TODO: we should test changing the step on other platforms.
+    // The general tradeoff is making extremely large strings potentially much faster
+    // at the cost of small strings being slightly slower.
+    const step = size;
+    var i: usize = 0;
+    while (i + step < buf.len) {
+        var bytes: usize = 0;
+        @memcpy(@ptrCast([*]u8, &bytes), @ptrCast([*]const u8, buf) + i, size);
```

Comment:
Since this is only copying a single word, would it make sense to cast `buf` to `*usize`, dereference that, and assign it to `bytes`, instead of using `memcpy`?
Maybe there's an alignment issue there and Zig forces you to use `memcpy`

Reply:
Yes, exactly that. Because it may be unaligned we have to use `memcpy` or manually copy the bytes in some from. In reality, the compiler sees what you are doing and fully removes the `memcpy`, emitting an unaligned load.